students/dowoon/users/validators.py

```diff
@@ -0,0 +1,20 @@
+import re
+
+from users.models import User
+
+
+def validate_email(email):
+    if not re.match("^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$", email):
+        raise Exception("ERROR : INVALID_VALUE (email)")
+
+def validate_password(password):
+    if not re.match("^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[$@$!%*?&])[A-Za-z\d$@$!%*?&]{8,}", password):
+        raise Exception("ERROR : INVALID_VALUE (password)")
+
+def validate_duplicate(email):
+    try:
+        user = User.objects.get(email=email)
+    except User.DoesNotExist:
+        if user.email == email:
+            raise Exception("ERROR : DUPLICATE")
+
```

Comment:
피드백을 몇가지 드리자면,
1. `validate_duplicate` 함수명은 정확히 어떤 값의 중복을 검증하는 함수인지 명확하게 알기 어렵습니다.
2. validate_duplicate 함수의 로직이 이상해보입니다. `user = User.objects.get(email=email)`가 실행되는 부분에서 에러가 발생되지 않는 다는 의미는 이미 데이터에비스에 유저가 존재한다는 의미인데, 위 로직에서는 반대로 error가 발생할 때 중복되었다고 에러를 발생시키고 있습니다.
3. 
아래와 같이 exists() 함수를 사용해서 True or False 값으로 존재 여부를 판단할 수 있도록 수정하는 것이 좋겠습니다.

```suggestion
def validate_duplicate(email):
    if not User.objects.filter(email=email).exists():
        return False #혹은 raise Exception ..
        
    return True
```

Reply:
확인 / 수정했습니다.